microcosm_pubsub/chain/statements.py

```diff
@@ -0,0 +1,133 @@
+from marshmallow import ValidationError
+
+
+class ArgumentExtractor:
+    def __init__(self, name, key, key_property=None):
+        self.name = name
+        self.key = key
+        self.key_property = key_property
+
+    def __str__(self):
+        return f"extract_{self.name}"
+
+    def __call__(self, context):
+        if self.name in context:
+            raise ValidationError(f"Variable '{self.name}'' alredy extracted")
+
+        obj = context[self.key]
+        if self.key_property is None:
+            value = obj
+        elif hasattr(obj, self.key_property):
+            value = getattr(obj, self.key_property)
+        else:
+            value = obj[self.key_property]
+
+        context[self.name] = value
+        return value
+
+
+class IfStatement:
+    def __init__(self, name, chain=None, other=None):
+        self.name = name
+        self.chain = chain
+        self.other = other
+
+    def __str__(self):
+        return f"if_{self.name}"
+
+    def __call__(self, context):
+        if context[self.name]:
+            if self.chain:
+                return self.chain(context)
+        elif self.other:
+            return self.other(context)
+
+
+class SwitchStatement:
+    def __init__(self, name, cases, other=None):
+        self.name = name
+        self.other = other
+        self.cases = cases
+
+    def __str__(self):
+        return f"switch_{self.name}"
+
+    def __call__(self, context):
+        key = context[self.name]
+        action = self.cases.get(key, self.other)
+        if action:
+            return action(context)
+
+
+class TryStatement:
+    def __init__(self, chain, catch, other=None):
+        self.chain = chain
+        self.catch = catch
+        self.other = other
+
+    def __call__(self, context):
+        try:
+            res = self.chain(context)
+        except Exception as excption:
+            handle = self.catch.get(type(excption))
+            if not handle:
+                raise excption
+            return handle(context)
+        else:
+            if self.other:
+                return self.other(context)
+            return res
+
+
+def extract(name, key, key_property=None):
+    """
+    Extract an argument from a context to another context key
+
+    :param name: new context key
+    :param from_: old context key
+    :param property_: propery of the context key
+
+    """
+    return ArgumentExtractor(name, key, key_property)
+
+
+def if_(name, chain=None, other=None):
```

Comment:
I've tried to avoid python keywords such as `if`, `else`, `try`...
I didn't find something that I like that can replace `if_`,`condition` was the closest.  Better idea?

Reply:
Maybe `when` ?